docs/diagrams/DeadlineState2.puml

```diff
@@ -0,0 +1,18 @@
+@startuml
+!include style.puml
+skinparam arrowThickness 1.1
+skinparam arrowColor LOGIC_COLOR_T4
+skinparam classBackgroundColor LOGIC_COLOR
+
+package List<Person> {
+Class "List[index - 1]" as Person1
+Class "List[index]" as Person2
+Class "List[index + 1]" as Person3
+}
+
+Person1 - Person2
+Person2 - Person3
+
+class Pointer as "personToAddDeadline"
+Pointer -up-> Person2
```

Comment:
To add to this, since the DeadlineCommand also has a reference to editedPerson, I think you can have it point to editedPerson in diagram 3!

Reply:
I mainly just wanted to convey that the program creates a different person and replaces the existing person. I think what you suggested works, and I hope it will bring more clarity.